python/ct/crypto/ecdsa_keys.py

```diff
@@ -0,0 +1,76 @@
+from ct.crypto import error
+from ct.crypto import pem
+from ct.proto import client_pb2
+
+import hashlib
+import ecdsa
+
+class EcdsaPublicKey(object):
+    """Wraps an ECDSA public key."""
+
+    # The signature algorithm used for this public key."""
+    SIGNATURE_ALGORITHM = client_pb2.DigitallySigned.ECDSA
+    # The hash algorithm used for this public key."""
+    HASH_ALGORITHM = client_pb2.DigitallySigned.SHA256
+
+    # Markers to look for when reading a PEM-encoded ECDSA public key."""
+    __READ_MARKERS = ("PUBLIC KEY", "ECDSA PUBLIC KEY")
+    # A marker to write when writing a PEM-encoded ECDSA public key."""
+    __WRITE_MARKER = "ECDSA PUBLIC KEY"
+
+    def __init__(self, key_info):
+        """Loads a PEM-encoded ECDSA public key.
+
+        Args:
+        - key_info: KeyInfo protobuf message
+
+        Raises:
+        - PemError: If the key has an invalid encoding
+        """
+        if (key_info.type != client_pb2.KeyInfo.ECDSA):
+            raise error.UnsupportedAlgorithmError(
+                "Expected ECDSA key, but got key type %d" % key_info.type)
+
+        # Will raise a PemError on invalid encoding
+        self.__der, _ = pem.from_pem(key_info.pem_key, self.__READ_MARKERS)
+        try:
+            self.__key = ecdsa.VerifyingKey.from_der(self.__der)
+        except ecdsa.der.UnexpectedDER as e:
+            raise error.EncodingError(e)
+
+    def __repr__(self):
+        return "%r(public key: %r)" % (self.__class__.__name__,
+                                       pem.to_pem(self.__der,
+                                                  self.__WRITE_MARKER))
+
+    def __str__(self):
```

Comment:
These two methods are subtlety different (one uses %s, the other %r). Nonetheless, I could extract their bulk to a private method. This is also the case in LogVerifier, so I'll do the same there too.